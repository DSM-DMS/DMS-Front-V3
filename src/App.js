import React, { Component, Fragment } from 'react';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
import './App.scss';
import axios from 'axios';

import { connect } from 'react-redux';
import { autoLogin, isLogin } from './actions';
import { getCookie, setCookie, removeCookie } from './lib/cookie';

import MainContainer from './student/container/Main/MainContainer';
import ApplyMainContainer from './student/container/Apply/ApplyMainContainer';
import StudentDefaultLayout from './student/common/DefaultLayout/DefaultLayout';
import MyPageContainer from './student/container/MyPage/MyPageContainer';
import GuideMainContainer from './student/container/Guide/GuideMainContainer';
import DevelopersContainer from './student/container/Developers/DevelopersContainer';
import FixContainer from './admin/container/Fix/FixContainer';
import FixDetailContainer from './admin/container/Fix/FixDetailContainer';
import SurveyContainer from './admin/container/Survey/SurveyContainer';
import SurveyWriteContainer from './admin/container/Survey/SurveyWriteContainer';
import LoginConatiner from './admin/container/Login/LoginContainer';
import ShowNoticeContainer from './admin/container/Notice/ShowNoticeContainer';
import GoingOutContainer from './admin/container/GoingOut/GointOutContainer';
import DomitoryRuleContainer from "./admin/container/DomitoryRule/DomitoryRuleContainer"
import DomitoryRuleWriteContainer from "./admin/container/DomitoryRule/DomitoryRuleWriteContainer"
import DomitoryRuleEditContainer from "./admin/container/DomitoryRule/DomitoryRuleEditContainer"
import CommonDesign from "./admin/common/commonDesign/CommonDesign"
import ShowDomitoryRuleContainer from "./admin/container/DomitoryRule/ShowDomitoryRuleContainer"
import NoticeContainer from './admin/container/Notice/NoticeContainer';
import NoticeWriteContainer from './admin/container/Notice/NoticeWriteContainer';
import NoticeEditContainer from './admin/container/Notice/NoticeEditContainer';
import AdminMainContainer from "./admin/container/Main/AdminMainContainer";
import MusicContainer from "./admin/container/Music/MusicContainer"

import setHeader from './lib/setHeader';

axios.interceptors.request.use(
  conf => {
    if(window.location.href.includes('admin') === true) {
      return conf;
    }
    conf.headers = setHeader(conf.headers);
    return conf;
  },
  err => {
    return Promise.reject(err);
  },
);

axios.interceptors.response.use(
  res => {
    if (!!res.headers['new-access-token']) {
      setCookie('JWT', res.headers['new-access-token']);
    }
    return res;
  },
  err => {
    return Promise.reject(err);
  },
);

class App extends Component {
  componentWillMount() {
    if (
      (navigator.appName === 'Netscape' &&
        navigator.userAgent.search('Trident') !== -1) ||
      navigator.userAgent.indexOf('msie') !== -1
    ) {
      alert(
        '인터넷 익스플로러 브라우저 입니다.\n DMS는 공식적으로 IE를 지원하지 않습니다.',
      );
    }

    this.loginApi();

    // window.addEventListener('load', this.loginApi);
    window.addEventListener('beforeunload', this.setAutoLogin);
  }

  loginApi = () => {
    const id = getCookie('id');
    const pw = getCookie('pw');
    if (id && pw) {
      axios
        .post('https://dms-api.istruly.sexy/account/auth', {
          id: id,
          password: pw,
        })
        .then(response => {
          if (response.status === 200) {
            setCookie('JWT', response.data.accessToken);
            setCookie('ID', id);
            removeCookie('id');
            removeCookie('pw');
            this.props.autoLogin({ id: id, pw: pw });
            this.props.isLogin(true);
          } else if (response.status === 204) {
            alert('비밀번호가 틀렸습니다.');
          }
        })
        .catch(err => {
          removeCookie('id');
          removeCookie('pw');
        });

      // if (response.status === 200) {
      //   setCookie('JWT', response.data.accessToken);
      //   setCookie('ID', id);
      //   removeCookie('id');
      //   removeCookie('pw');
      //   this.props.autoLogin({ id: id, pw: pw });
      //   this.props.isLogin(true);
      // }
    }
  };

  setAutoLogin = () => {
    const { id, pw, autoLogin } = this.props;
    if (id && pw) {
      autoLogin({ id: '', pw: '' });
      setCookie('id', id, 150);
      setCookie('pw', pw, 150);
    }
    removeCookie('JWT');
    removeCookie('ID');
  };

  render() {
    if (
      (navigator.appName === 'Netscape' &&
        navigator.userAgent.search('Trident') !== -1) ||
      navigator.userAgent.indexOf('msie') !== -1
    ) {
      return (
        <div
          style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            fontSize: '3rem',
            color: 'red',
            height: '100vh',
          }}
        >
          IE를 지원하지 않습니다. 타 브라우저로 접속해주세요.
        </div>
      );
    }
    return (
      <BrowserRouter>
        <Switch>
          <Route path="/admin" component={AdminMainContainer} exact />
          <Route path="/apply/extension" component={ApplyMainContainer} exact />
          <Route path="/apply/goingout" component={ApplyMainContainer} exact />
          <Route path="/apply/stay" component={ApplyMainContainer} exact />
          <Route path="/apply/music" component={ApplyMainContainer} exact />
          <Route path="/guide/faq" component={GuideMainContainer} exact />
          <Route path="/guide/notice" component={GuideMainContainer} exact />
          <Route path="/guide/rule" component={GuideMainContainer} exact />
          <Route path="/developers" component={DevelopersContainer} exact />
          <Route path="/admin/login" component={LoginConatiner} exact/>
          <Route
            path="/admin/:uri?"
            render={() => (
              <CommonDesign>
                  <Switch>
                    <Route path="/admin/domitoryrule" component={DomitoryRuleContainer} exact/>
                    <Route path="/admin/domitoryrule/write" component={DomitoryRuleWriteContainer} exact />
                    <Route path="/admin/domitoryrule/edit/:postId" component={DomitoryRuleEditContainer} exact />
                    <Route path="/admin/domitoryrule/:postId" component={ShowDomitoryRuleContainer}/>
                    <Route path="/admin/notice" component={NoticeContainer} exact/>
                    <Route path="/admin/notice/write" component={NoticeWriteContainer} exact/>
                    <Route path="/admin/notice/edit/:postId" component={NoticeEditContainer} exact />
                    <Route path="/admin/notice/:postId" component={ShowNoticeContainer}/>
                    <Route path="/admin/fix" component={FixContainer} exact/>
                    <Route path="/admin/fix/:uri" component={FixDetailContainer} exact/>
                    <Route path="/admin/goingout" component={GoingOutContainer} exact/>
                    <Route path="/admin/music" component={MusicContainer} exact/>
                    <Redirect to="/admin/login" />
                   </Switch>
              </CommonDesign>
            )}
          />
          <Route
            path="/:uri?"
            render={() => (
              <Fragment>
                <StudentDefaultLayout>
                  <Switch>
                    <Route path="/" component={MainContainer} exact />
                    <Route path="/apply" component={MainContainer} exact />
                    <Route path="/guide" component={MainContainer} exact />
                    <Route path="/extra" component={MainContainer} exact />
                    <Route path="/mypage" component={MyPageContainer} exact />
                  </Switch>
                </StudentDefaultLayout>
              </Fragment>
            )}
          />
          }/>
          <Redirect to="/" />
        </Switch>
      </BrowserRouter>
    );
  }
}

const mapStateToProps = state => ({
  id: state.login.id,
  pw: state.login.pw,
});

const mapDispatchToProps = dispatch => ({
  autoLogin: val => dispatch(autoLogin(val)),
  isLogin: bool => dispatch(isLogin(bool)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(App);
