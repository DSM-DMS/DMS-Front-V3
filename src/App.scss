.App {
  text-align: center;
}

@font-face {
  font-family: 'NanumSquareEB';
  src: url('./assets/font/NANUMSQUAREEB.TTF') format('truetype')
}


@font-face {
  font-family: 'NanumSquareB';
  src: url('./assets/font/NANUMSQUAREB.TTF') format('truetype')
}

@font-face {
  font-family: 'NanumSquareL';
  src: url('./assets/font/NANUMSQUAREL.TTF') format('truetype')
}

@font-face {
  font-family: 'NanumSquareR';
  src: url('./assets/font/NANUMSQUARER.TTF') format('truetype')
}