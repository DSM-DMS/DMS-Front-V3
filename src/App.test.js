import React from "react";
import { shallow } from "enzyme";

import App from "./App";

describe("App", () => {
  let component = null;
  it("should renders correctly", () => {
    component = shallow(<App />);
  });

  it("should matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
