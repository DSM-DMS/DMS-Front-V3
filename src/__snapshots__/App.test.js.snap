// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(ConnectedRouterWithContext)
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/admin"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/extension"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/goingout"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/stay"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/music"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/mypage"
            />
            <Route
              path="/admin/:uri?"
              render={[Function]}
            />
            <Route
              path="/:uri?"
              render={[Function]}
            />
            }/&gt;
            <Redirect
              push={false}
              to="/"
            />
          </Switch>
        </BrowserRouter>
      </Connect(ConnectedRouterWithContext)>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/admin"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/extension"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/goingout"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/stay"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/music"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/mypage"
            />
            <Route
              path="/admin/:uri?"
              render={[Function]}
            />
            <Route
              path="/:uri?"
              render={[Function]}
            />
            }/&gt;
            <Redirect
              push={false}
              to="/"
            />
          </Switch>
        </BrowserRouter>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/admin"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/extension"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/goingout"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/stay"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/apply/music"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/mypage"
            />
            <Route
              path="/admin/:uri?"
              render={[Function]}
            />
            <Route
              path="/:uri?"
              render={[Function]}
            />
            }/&gt;
            <Redirect
              push={false}
              to="/"
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/admin"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/apply/extension"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/apply/goingout"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/apply/stay"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/apply/music"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/mypage"
              />,
              <Route
                path="/admin/:uri?"
                render={[Function]}
              />,
              <Route
                path="/:uri?"
                render={[Function]}
              />,
              "}/>",
              <Redirect
                push={false}
                to="/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/admin",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/apply/extension",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/apply/goingout",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/apply/stay",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/apply/music",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/mypage",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/admin/:uri?",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/:uri?",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "}/>",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "push": false,
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(ConnectedRouterWithContext)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <BrowserRouter>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/admin"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/extension"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/goingout"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/stay"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/music"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/mypage"
              />
              <Route
                path="/admin/:uri?"
                render={[Function]}
              />
              <Route
                path="/:uri?"
                render={[Function]}
              />
              }/&gt;
              <Redirect
                push={false}
                to="/"
              />
            </Switch>
          </BrowserRouter>
        </Connect(ConnectedRouterWithContext)>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <BrowserRouter>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/admin"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/extension"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/goingout"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/stay"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/music"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/mypage"
              />
              <Route
                path="/admin/:uri?"
                render={[Function]}
              />
              <Route
                path="/:uri?"
                render={[Function]}
              />
              }/&gt;
              <Redirect
                push={false}
                to="/"
              />
            </Switch>
          </BrowserRouter>,
          "history": Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/admin"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/extension"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/goingout"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/stay"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/apply/music"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/mypage"
              />
              <Route
                path="/admin/:uri?"
                render={[Function]}
              />
              <Route
                path="/:uri?"
                render={[Function]}
              />
              }/&gt;
              <Redirect
                push={false}
                to="/"
              />
            </Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/admin"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/apply/extension"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/apply/goingout"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/apply/stay"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/apply/music"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/mypage"
                />,
                <Route
                  path="/admin/:uri?"
                  render={[Function]}
                />,
                <Route
                  path="/:uri?"
                  render={[Function]}
                />,
                "}/>",
                <Redirect
                  push={false}
                  to="/"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/admin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/apply/extension",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/apply/goingout",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/apply/stay",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/apply/music",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/mypage",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/admin/:uri?",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/:uri?",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "}/>",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "push": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
